src/dialogs/quick-bar/ha-quick-bar.ts

```diff
@@ -181,16 +188,37 @@ export class QuickBar extends LitElement {
     }
 
     this._items = (this._commandMode ? this._commandItems : this._entityItems)
-      .filter(({ text, altText }) => {
-        const values = [text];
-        if (altText) {
-          values.push(altText);
-        }
-        return fuzzySequentialMatch(this._itemFilter, values);
-      })
-      .sort((itemA, itemB) => compare(itemA.text, itemB.text));
+      .filter(({ text, altText }) =>
+        altText
+          ? isPatternInWords(this._itemFilter, text, altText)
+          : isPatternInWords(this._itemFilter, text)
+      )
+      .map<ScoredItem>((item) => ({
+        item,
+        score: item.altText
+          ? fuzzySequentialMatch(this._itemFilter, item.text, item.altText)
+          : fuzzySequentialMatch(this._itemFilter, item.text),
+      }))
+      .sort(this.compareFuzzyMatches)
+      .map((scoredItem) => scoredItem.item);
   }
 
+  private compareFuzzyMatches = (a: ScoredItem, b: ScoredItem) => {
+    const scoreA = a.score;
+    const scoreB = b.score;
+
+    if (!scoreA || !scoreB) {
+      return 0;
+    }
+
+    const isNumber = !isNaN(Number(scoreA));
+    if (isNumber) {
+      return scoreA > scoreB ? -1 : 1;
+    }
+
+    return compare(a.item.text, b.item.text);
+  };
```

Comment:
Should we move this to `sequence-matching.ts`?

Reply:
We can. I'll just need to move `ScoredItem` interface there as well, and change it to use generics so it's not pulling in `QuickBarItem` to extend the type.

Something like this:
```
// sequence-matching.ts
interface ItemWithText {
  text: string;
}

export interface ScoredItem<T extends ItemWithText> {
  item: T;
  score: ReturnType<typeof fuzzySequentialMatch>;
}

//ha-quick-bar.ts
interface QuickBarItem extends ItemWithText {
  altText?: string;
  icon: string;
  action(data?: any): void;
}
```

Reply:
I guess I could do just:

```
export interface ScoredItem<T extends { text: string }> {
  item: T;
  score: ReturnType<typeof fuzzySequentialMatch>;
}
```

Reply:
None of this was needed. Resolving

Reply:
Turns out this was needed (it wasn't actually sorting by score and regressed to a simple "passes, but sorting alphabetically" behavior).

I've moved all that logic into sequence matcher so I could add tests to avoid this from happening in the future.